Add setting to allow InviteToken to override EnableAccountCreation

theit8514 opened the pull request:
Adds logic to SignupAsync to allow InviteTokens to override the EnableAccountCreation flag when the EnableInvitedUserCreation flag is set. Requested PR from issue #147.

Unit tests are modeled after existing controller tests but are flaky when run as a group because of xUnit's async task runner and the updating of the global Settings. Let me know if you want to trim this down to just the logic change.

ejsmith commented:
Would it be better to just always allow invite tokens to work even if account creation is turned off?

theit8514 commented:
Adding the token to the check there wasn't working. I split the two out.

theit8514 commented:
I'm okay with always allowing Invite Tokens to always work.

niemyjski commented:
Thanks for your work! If the tests are flakey I'd rather not introduce them as we already have a few rouge unit tests I've been trying to track down. 

I'd also prefer just using the token if it's valid to override the account creation process. This would also keep the amount of settings we have to a minimum.

theit8514 commented:
I've cleaned up the setting and squashed all the commits. The concurrency issue I was having was related to RemoveAllAsync on the users repository. With each test getting it's own email, they no longer interfere with each other.

niemyjski commented:
Thank you!
